docs/docs/guides/evaluation/guardrails_and_monitors.md

```diff
@@ -5,139 +5,482 @@ import TabItem from '@theme/TabItem';
+- `output`: The result from your function
+- Any input parameters matching your function's parameters
+
+Here's a comprehensive example:
 
 ```python
-async def monitored_operation(user_input: str, sampling_rate: float = 0.25) -> str:
-    """
-    Execute operation with monitoring.
-    
-    Args:
-        user_input: The input to process
-        sampling_rate: Percentage of operations to monitor (0.0 to 1.0)
+@weave.op()
+def generate_styled_text(prompt: str, style: str, temperature: float) -> str:
+    """Generate text in a specific style."""
+    return "Generated text in requested style..."
+
+class StyleScorer(Scorer):
+    @weave.op
+    def score(self, output: str, prompt: str, style: str) -> dict:
+        """
+        Evaluate if the output matches the requested style.
         
-    Returns:
-        str: Operation result
-    """
-    result, call = op.call(user_input)
-    
-    # Apply monitoring based on sampling rate
-    if random.random() < sampling_rate:
-        await call.apply_scorer(scorer)
+        Args:
+            output: The generated text (automatically provided)
+            prompt: Original prompt (matched from function input)
+            style: Requested style (matched from function input)
+        """
+        return {
+            "style_match": 0.9,  # How well it matches requested style
+            "prompt_relevance": 0.8  # How relevant to the prompt
+        }
+
+# Example usage
+async def generate_and_score():
+    # Generate text with style
+    result, call = generate_styled_text.call(
+        prompt="Write a story",
+        style="noir",
+        temperature=0.7
+    )
     
-    return result
+    # Score the result
+    score = await call.apply_scorer(StyleScorer())
+    print(f"Style match score: {score.score['style_match']}")
 ```
 
-:::caution Performance Considerations
-Scorer evaluations execute synchronously on the same machine as the operation. For high-throughput production environments, consider adjusting sampling rates based on load. Weave will soon support server-side scoring for high-throughput applications.
+:::tip Parameter Matching Rules
+- The `output` parameter is special and always contains the function's result
+- Other parameters must match the function's parameter names exactly
+- Scorers can use any subset of the function's parameters
+- Parameter types should match the function's type hints
 :::
 
-## Analysis and Observability
+:::info Handling Parameter Name Mismatches
+Sometimes your scorer's parameter names might not match your function's parameter names exactly. For example:
 
-### Accessing Scorer Results
+```python
+@weave.op()
+def generate_text(user_input: str):  # Uses 'user_input'
+    return process(user_input)
+
+class QualityScorer(Scorer):
+    @weave.op
+    def score(self, output: str, prompt: str):  # Expects 'prompt'
+        """Evaluate response quality."""
+        return {"quality_score": evaluate_quality(prompt, output)}
+
```

Comment:
Suggested adding this line here, just for consistency to show where `call` comes from:

```
result, call = generate_text.call(user_input = "Say hello")
```

Reply:
done